### DISCUSSION TITLE
How to auto start "pdm run read_dump_serve.py" on boot?

### DISCUSSION THREAD

**Discussion by nilsharm** [phiresky/ebv-theta-to-mqtt#5] [Q&A] [answered]
I created a bash script that is executed on boot.
The script creates a test file using the touch command and calls "pdm run read_dump_serve.py" (with absolute paths). When I start the script manually, everything is fine. However, executing on boot only creates the test file, but pdm does not start. 
Any ideas how to start the read_dump_serve.py script on boot?

Edit: I am running everything on a Rasperry Pi with Raspberry Pi OS Lite (32 Bit), Debian Bookworm.

**Comment by phiresky** [comment, marked as answer]
The systemd service file I use for auto start looks like this:

```
systemctl cat theta
# /etc/systemd/system/theta.service
[Unit]
Description=theta to mqtt
[Service]
ExecStart=/root/.local/bin/pdm run read_dump_serve.py --mqtt_hostname=homeassistant
Restart=on-failure
RestartSec=20
WorkingDirectory=/root/2023-01-theta-heizung
Environment=PYTHONUNBUFFERED=1


[Install]
WantedBy=multi-user.target
```

**Reply by phiresky** [reply]
The relevant part for your question is the WorkingDirectory=...
